tests/unit/Owncloud/OcisPhpSdk/GroupTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace unit\Owncloud\OcisPhpSdk;
+
+use OpenAPI\Client\Model\Group;
+use Owncloud\OcisPhpSdk\Exception\InvalidResponseException;
+use Owncloud\OcisPhpSdk\Group as SdkGroup;
+use PHPUnit\Framework\TestCase;
+
+class GroupTest extends TestCase
+{
+    private function validGroupData(): array
+    {
+        return [
+            [[
+                "id" => "id",
+                "description" => "desc",
+                "display_name" => "displayname",
+                "group_types" => ["aa"],
+                "members" => [],
+                // "membersodata_bind" => [],
+            ]],
+            [[
+                "id" => "id",
+                "description" => "desc",
+                "display_name" => "displayname",
+                "group_types" => [],
+                "members" => [],
+                // "membersodata_bind" => [],
+            ]],
+            [[
+                "id" => "id",
+                "description" => "desc",
+                "display_name" => "displayname",
+                "group_types" => [],
+                "members" => [],
+                // "membersodata_bind" => [],
+            ]],
+        ];}
+
+/**
+ * @dataProvider validGroupData
+ * @return void
+ */
+    public function testListGroup(array $data)
+    {
+        $libGroup = new Group($data);
+        $group = new SdkGroup($libGroup);
+        $this->assertInstanceOf(SdkGroup::class, $group);
+        $this->assertEquals($data["id"], $group->getId());
+        $this->assertEquals($data["description"], $group->getDescription());
+        $this->assertEquals($data["display_name"], $group->getDisplayName());
+        $this->assertSame($data["group_types"], $group->getGroupTypes());
+        $this->assertSame($data["members"], $group->getMembers());
+    }
+
+    private function InvalidGroupData(): array
+    {
+        return [
+            [[
+                "id" => null,
+                "description" => "",
+                "display_name" => "asd",
+                "group_types" => ["aa"],
+                "members" => [],
+            ]],
+            [[
+                "id" => "",
+                "description" => "a",
+                "display_name" => "as",
+                "group_types" => ["aa"],
+                "members" => [],
+            ]],
```

Comment:
we should comment on what is invalid in this case
```suggestion
            [[  // id is null
                "id" => null,
                "description" => "",
                "display_name" => "asd",
                "group_types" => ["aa"],
                "members" => [],
            ]],
            [[  // id is an empty string
                "id" => "",
                "description" => "a",
                "display_name" => "as",
                "group_types" => ["aa"],
                "members" => [],
            ]],
```